src/omnicore/tx.cpp

```diff
@@ -165,22 +183,38 @@ bool CMPTransaction::interpret_Transaction()
 /** Version and type */
 bool CMPTransaction::interpret_TransactionType()
 {
-    if (pkt_size < 4) {
-        return false;
+    if (encodingClass == OMNI_CLASS_D) {
+        int i = 0;
+
+        std::vector<uint8_t> vecVersionBytes = GetNextVarIntBytes(i);
```

Comment:
Is this safe in any case, e.g. without payload and only marker? If I'm not mistaken, `pkt[0]` is accessed in that case, too, and without payload, this results in an out of bounds array access?